tests/unit/test_service_layer_handlers.py

```diff
@@ -19,38 +20,69 @@
 
 import fingerprint_report_templates
 
-TEST_ENDPOINT = "http://localhost:3020/dev/query"
-OUTPUT_LOCATION = "./output"
-
 
 def test_accessing_templates():
     with pkg_resources.path(fingerprint_report_templates, "fingerprint_report") as resource_path:
         assert "fingerprint_report" in str(resource_path)
 
 
-def test_reading_template_confing_json_file():
+def test_reading_template_config_json_file():
     with pkg_resources.path(fingerprint_report_templates, "fingerprint_report") as resource_path:
         config_dict = json.loads((resource_path / "config.json").read_bytes())
         assert "conf" in config_dict
         assert "template" in config_dict
 
 
 def test_generate_report_builder_config():
-    d = generate_report_builder_config(sparql_endpoint_url="url", graph="graph")
+    d = generate_report_builder_config(sparql_endpoint_url="url", graph="graph", external_template_location=None)
     assert "url" == d["conf"]["default_endpoint"]
     assert "graph" == d["conf"]["default_graph"]
 
 
-def test_generate_endpoint_fingerprint_report():
+def test_generate_endpoint_fingerprint_report_default_template(tmpdir):
     """
         given  an input endpoint
         when generate is executed
         the output folder has a report file
     :return:
     """
-    output_location = Path(OUTPUT_LOCATION)
-    shutil.rmtree(output_location, ignore_errors=True)
-    output_location.mkdir(parents=True, exist_ok=True)
-    output_file = generate_endpoint_fingerprint_report(sparql_endpoint_url=TEST_ENDPOINT,
-                                                       output_location=OUTPUT_LOCATION)
-    assert output_file.exists()
+    output_path = tmpdir.mkdir("/output")
+
+    output_file = generate_endpoint_fingerprint_report(sparql_endpoint_url='http://localhost:3030/dev/query',
+                                                       output_location=str(output_path))
+
+    with open(output_file, 'r') as file:
+        soup = BeautifulSoup(file.read(), 'html.parser')
+
+    main_title = soup.find('h1', attrs={'class': 'ui header center aligned reportTitle'}).text
+    assert main_title == 'Structural fingerprint'
+
+    class_instantiation_title = soup.find('h1', attrs={'class': 'ui header'}).text
+    assert class_instantiation_title == 'Class instantiation shapes'
+
+    tables = soup.find_all('table')
+    assert len(tables) == 6
+
+    classes_list = ['owl#Ontology', 'core#ConceptScheme', 'skos-xl#Label',
+                    'core#Concept', 'euvoc#Continent', 'euvoc#XlNotation']
+
+    def check_if_class_in_report(class_name, table_list):
```

Comment:
the logic of this function could have been written in the for loop below, but fine, now we have a funky test :)

Reply:
:laughing: , right, was blind-sighted because I started the implementation in another way.